This patch fixes a Solaris specific issue where ifoperstatus of
the vnic's inside a non global zone and also ifoperstatus of
the vlan shows it as down even though the link is up. The patch
has been submitted to the community but has not been accepted yet.

The details can be found in the following location
https://sourceforge.net/p/net-snmp/bugs/2605/

diff -wpruN '--exclude=*.orig' net-snmp-5.8~/agent/mibgroup/kernel_sunos5.c net-snmp-5.8/agent/mibgroup/kernel_sunos5.c
--- net-snmp-5.8~/agent/mibgroup/kernel_sunos5.c	2018-07-24 09:51:43.470214338 +0000
+++ net-snmp-5.8/agent/mibgroup/kernel_sunos5.c	2018-07-24 09:51:44.648186134 +0000
@@ -1963,8 +1963,8 @@ set_if_info(mib2_ifEntry_t *ifp, unsigne
     if (ifp->ifAdminStatus == 1) {
         int i_tmp;
         /* only UPed interfaces get correct link status - if any */
-        if (getKstatInt(NULL, name,"link_up",&i_tmp) == 0) {
-            ifp->ifOperStatus = i_tmp ? 1 : 2;
+        if (getKstatInt("link", name,"phys_state",&i_tmp) == 0) {
+            ifp->ifOperStatus = (i_tmp == 1) ? 1 : ((i_tmp == 0) ? 2 : 4);
         } else if(link_to_phydev(name, phydevname) == 0) {
                  strcpy(devbasename, phydevname);
                  DEBUGMSGTL(("kernel_sunos5","phydevname = %s\n", phydevname));
